### PR TITLE
BSS - deal updated timestamp on the top level, aligned with GEF

### PR DESCRIPTION
**Pull request by ttbarnes**
- change `deal.details.dateOfLastAction` to be `deal.updatedAt`
- use `Date.now()` instead of moment
- rename some field names used in GraphQL/Mongo queries/mapping, for Portal Dashboard. Now the timestamp field has the same name as the actual data.
- also moved some incorrect `submissionType` fields in Swagger docs (missed this in last PR)

### CODE REVIEW

File: e2e-tests/portal/cypress/integration/journeys/maker/about-supply-contract/dealWithSecondPageComplete.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
     "submissionType": "Automatic Inclusion Notice",
+    "updatedAt": 1639675204629.0,
```
Comment (abhi-markan) [inline e2e-tests/portal/cypress/integration/journeys/maker/about-supply-contract/dealWithSecondPageComplete.json:3]: Decimal, is it required?

File: portal-api/api-tests/v1/deals/deal-name.api-test.js [hunk 2]
```diff
@@ -7,11 +7,12 @@ const testUserCache = require('../../api-test-users');
 const { as } = require('../../api')(app);
 
 const newDeal = aDeal({
+  updatedAt: Date.now(),
   details: {
     bankSupplyContractName: 'mock name',
     bankSupplyContractID: 'mock id',
+    status: "Draft",
```
Comment (IgnatG) [inline portal-api/api-tests/v1/deals/deal-name.api-test.js:14]: this seems to be duplicated below

File: portal-api/api-tests/fixtures/deal-with-complete-about-section.json [hunk 3]
```diff
@@ -1,11 +1,11 @@
 {
+    "submissionType": "Automatic Inclusion Notice",
+    "updatedAt": 1639675204629.0,
```
Comment (abhi-markan) [inline portal-api/api-tests/fixtures/deal-with-complete-about-section.json:3]: Decimal?

File: portal-api/api-tests/fixtures/deal-with-incomplete-about-section.json [hunk 4]
```diff
@@ -1,11 +1,11 @@
 {
+    "submissionType": "Automatic Inclusion Notice",
+    "updatedAt": 1639675204629.0,
```
Comment (abhi-markan) [inline portal-api/api-tests/fixtures/deal-with-incomplete-about-section.json:3]: Decimal?

File: e2e-tests/portal/cypress/integration/journeys/maker/about-supply-contract/dealWithFirstPageComplete.json [hunk 5]
```diff
@@ -1,10 +1,10 @@
 {
     "submissionType": "Automatic Inclusion Notice",
+    "updatedAt": 1639675204629.0,
```
[the remark this answers is not in the document]
Comment (ttbarnes) [reply e2e-tests/portal/cypress/integration/journeys/maker/about-supply-contract/dealWithFirstPageComplete.json:3]: Was just a copy and paste I think

File: portal-api/api-tests/v1/deals/deal-name.api-test.js (hunk 2 above)
Comment (abhi-markan) [inline portal-api/api-tests/v1/deals/deal-name.api-test.js]: Status appearing twice?

File: portal-api/src/graphql/schemas/index.js [hunk 6]
```diff
@@ -52,6 +51,7 @@ type Deal {
   _id: String!
   dealType: String
   submissionType: String
+  updatedAt: Float
```
Comment (abhi-markan) [inline portal-api/src/graphql/schemas/index.js:54]: Float or Integer, if decimal is not required. If required please ignore.

File: portal-api/src/graphql/schemas/index.js [hunk 7]
```diff
@@ -61,8 +61,8 @@ type AllDeal {
   bankRef: String
   exporter: String
   product: String
-  type: String
-  lastUpdate: Float
+  submissionType: String
+  updatedAt: Float
```
Comment (abhi-markan) [inline portal-api/src/graphql/schemas/index.js:65]: Same as above. Thanks

File: portal-api/api-tests/v1/deals/deal-name.api-test.js (hunk 2 above)
Comment (ttbarnes) [reply portal-api/api-tests/v1/deals/deal-name.api-test.js]: Fixed :)

File: trade-finance-manager-api/README.md [hunk 8]
```diff
@@ -151,9 +151,9 @@ Therefore in TFM, we currently need to deal with 2 different data structures for
   _id: '1',
   dealType: 'BSS/EWCS',
   submissionType: 'Automatic Inclusion Notice',
+  updatedAt: 1639580324306.0
```
Comment (abhi-markan) [inline trade-finance-manager-api/README.md:154]: Decimal?

File: trade-finance-manager-api/src/graphql/schemas/index.js [hunk 9]
```diff
@@ -359,6 +358,7 @@ type DealSnapshot {
   _id: String!
   dealType: String!
   submissionType: String
+  updatedAt: Float
```
Comment (abhi-markan) [inline trade-finance-manager-api/src/graphql/schemas/index.js:361]: Float or Integer please?

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: portal-api/api-tests/v1/deals/deal-name.api-test.js (hunk 2 above)
Comment (ttbarnes) [reply portal-api/api-tests/v1/deals/deal-name.api-test.js]: Updated this

File: portal-api/src/graphql/schemas/index.js (hunk 6 above)
Comment (ttbarnes) [reply portal-api/src/graphql/schemas/index.js:54]: ATM sometimes it generates a decimal depending on the time

**Comment by ttbarnes** [comment]
The timestamp can have decimals - seems to depend on when exactly it's generated. Some GEF fields have the same, where in the DB the timestamp has a decimal. So leaving as Float type is fail safe.

**Review by abhi-markan — APPROVED** [review]

**Review by IgnatG — APPROVED** [review]

**Pull request merged**